src/placeos-compiler/git.cr

```diff
@@ -69,7 +69,7 @@ module PlaceOS::Compiler
       # %an: author name
       # %s: subject
       path = repository_path(repository, working_directory)
-      result = file_operation(path, file_name) do
+      result = repository_lock(path).read do
         run_git(path, {"fetch", "--all"})
```

Comment:
May be worth double checking this one. This `fetch` operation will be mutating the repository state and could lead to some unexpected behaviour if attempted for concurrent requests.